arangod/Aql/Ast.cpp

```diff
@@ -598,24 +598,17 @@ AstNode* Ast::createNodeCollect(AstNode const* groups, AstNode const* aggregates
   return node;
 }
 
-/// @brief create an AST collect node, COUNT INTO
+/// @brief create an AST collect node with a single COUNT aggregator
 AstNode* Ast::createNodeCollectCount(AstNode const* list, char const* name,
                                      size_t nameLength, AstNode const* options) {
-  AstNode* node = createNode(NODE_TYPE_COLLECT_COUNT);
-  node->reserve(2);
+  auto args = createNodeArray(1);
+  args->addMember(createNodeValueNull());
```

Comment:
Is it necessary for the COUNT/LENGTH to have a `null` input variable? Or will it even work without?

Reply:
At the moment all aggregator functions require exactly one argument, even though COUNT/LENGTH simply ignore it. I am planning to improve that, but at the moment we have to pass _something_.

Reply:
The `Aggregator` class has a static method `requiresInput` that we could leverage here or in other places that fail if no input argument is used:
```
102   /// @brief whether or not the aggregator requires any input or if the input
103   /// can be optimized away (note current: COUNT/LENGTH don't, all others do)
104   static bool requiresInput(std::string const& type);
```
This will return false for COUNT/LENGTH, and true for all other aggregate functions.